app/services/AtsListService.scala

```diff
@@ -94,26 +96,100 @@ class AtsListService @Inject() (
   } flatMap identity
 
   private def getAtsListAndStore(
-    agentToken: Option[AgentToken] = None
-  )(implicit hc: HeaderCarrier, request: AuthenticatedRequest[_]): Future[Either[AtsResponse, AtsListData]] = {
-    val account      = getAccount(request)
+                                  agentToken: Option[AgentToken] = None
+                                )(implicit hc: HeaderCarrier, request: AuthenticatedRequest[_]): Future[Either[AtsResponse, AtsListData]] = {
+    val account = getAccount(request)
     val requestedUTR = authUtils.getRequestedUtr(account, agentToken)
 
     val response = (account: @unchecked) match {
-      case agent: Uar        => middleConnector.connectToAtsListOnBehalfOf(agent, requestedUTR)
-      case individual: SaUtr => middleConnector.connectToAtsList(individual)
+      case agent: Uar => middleConnector.connectToAtsListOnBehalfOf(agent, requestedUTR)
+      case individual: SaUtr =>
+        var atsIndividualYearsCombinedList = List[AtsData]()
+        var taxYearsAvailableInIndividualYearsList = List[Int]()
+        var taxPayerDataFromIndividualYearsService = None: Option[TaxpayerFrontTierData]
+        val taxYears = atsTaxYearsUtils.getTaxYears
+        var totalFailuresInIndividualYearsApiCalls = 0
+        var yearFailureOccurredFor = 0
+        var dataNotFoundForAllYears = true
+        for (taxYearValue <- taxYears) {
+          print("sandeep" + taxYearValue)
+          val taxYearAtsDataForSingleYear = middleConnector.connectToAts(individual, taxYearValue)
+          if (taxYearAtsDataForSingleYear.value.nonEmpty) {
+            taxYearAtsDataForSingleYear.value.get.get match {
+
+              case AtsSuccessResponseWithPayload(data: AtsData) =>
+                dataNotFoundForAllYears = false
+                taxYearsAvailableInIndividualYearsList = data.taxYear :: taxYearsAvailableInIndividualYearsList
+                atsIndividualYearsCombinedList = data :: atsIndividualYearsCombinedList
+                taxPayerDataFromIndividualYearsService = Some(
+                  new TaxpayerFrontTierData(data.taxPayerData.get.taxpayer_name, None)
```

Comment:
Change the `taxPayerData.get` to a flatMap

`data.taxPayerData.flatMap(_.taxpayer_name)`